pkg/utils/tekton/controller.go

```diff
@@ -96,31 +139,66 @@ func (s *SuiteController) ListTaskRuns(ns string, labelKey string, labelValue st
 	return s.PipelineClient().TektonV1beta1().TaskRuns(ns).List(context.TODO(), listOptions)
 }
 
-func (k KubeController) WatchTaskPod(tr string, taskTimeout int) error {
-	trUpdated, err := k.Tektonctrl.GetTaskRun(tr, k.Namespace)
-	if err != nil {
-		return err
-	}
-	pod, err := k.Commonctrl.GetPod(k.Namespace, trUpdated.Status.PodName)
-	if err != nil {
-		return err
-	}
-	return k.Commonctrl.WaitForPod(k.Commonctrl.IsPodSuccessful(pod.Name, k.Namespace), time.Duration(taskTimeout)*time.Second)
+func (k KubeController) WatchPipelineRun(pipelineRunName string, taskTimeout int) error {
+	return k.Commonctrl.WaitForPod(k.Tektonctrl.CheckPipelineRunFinished(pipelineRunName, k.Namespace), time.Duration(taskTimeout)*time.Second)
```

Comment:
`WaitForPod` reads a little odd now in this context because a PipelineRun may have multiple pods. I realize now that this method is misnamed. It's actually just waiting for any condition function, which is what `CheckPipelineRunFinished` return. 

Maybe something we can improve in a future PR.

Reply:
Need to rebase anyhow, so I can touch upon this.